### DISCUSSION TITLE
Pyright maintained long-term?

### DISCUSSION THREAD

**Discussion by stoddart** [microsoft/pyright#2063] [Q&A]
In the FAQ, I read that Pyright "...will continue to be developed and maintained as an open-source project under its original MIT license terms. The Pyright extension for VSCode is a reference implementation and is not guaranteed to be fully functional or maintained long-term."

What does this mean? It seems contradictory, suggesting that Pyright will continue to be maintained and then that it's not guaranteed to be maintained.

**Comment by erictraut** [comment]
Pyright is currently published as both a command-line tool (in the npm registry) and a VS Code extension (in the VS Code Marketplace). We also now publish Pylance as a VS Code extension in the VS Code Marketplace. Since the Pylance extension offers a superset of Pyright's functionality, we may eventually stop publishing Pyright as a VS Code extension. If we were to do that, we would continue to publish Pyright as a command-line tool and as a language server that works with editors other than VS Code.

**Reply by stoddart** [reply]
Thanks for the explanation and taking the time to write this clarification; now I understand how Pylance and Pyright are related.
